### PR TITLE
Add native assignment editor

### PR DESCRIPTION
**Pull request by vanwagonet**
note that edits won't be immediately visible in the RN details & list views.
I didn't spend time fixing that since I intend to replace them soon with native ones that will pick up the CoreData changes automatically.

refs: MBL-14859
affects: Teacher
release note: none

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


[MBL-14859](https://instructure.atlassian.net/browse/MBL-14859)

#### Affected Apps: Teacher
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by inst-danger** [comment]
<details><summary>Student.ipa</summary>
  <a href="https://app.bitrise.io/artifact/55506528/p/28a1e65e2cec3238c68227dd32dbee52">
    
  </a>
</details>
<details><summary>Teacher.ipa</summary>
  <a href="https://app.bitrise.io/artifact/55506534/p/db66179669c7f3bf52d4a99e4430361d">
    
  </a>
</details>
<details><summary>Parent.ipa</summary>
  <a href="https://app.bitrise.io/artifact/55506485/p/6bda7e82023f94ec421febcf42808ae6">
    
  </a>
</details>

**Comment by balintbartok** [comment]
code looks good, functionality is perfect, just a few minor differences between the 2 versions, they are visible on the screenshot: size, section names, I don't know if it's intentional

**Comment by vanwagonet** [comment]
Yeah I named the "Options" section to match DiscussionEditorView. The description editor also grows in the new version. So the size differences are intentional. The screenshot didn't come through, so I hope I understood correctly.

**Comment by vanwagonet** [comment]
Let me know if the differences are acceptable to you, or if you'd rather match exactly what we had before. Personally, I much prefer the growing description editor so there is only a single scrolling view for the whole form.

**Comment by balintbartok** [comment]
not sure what happened with the screenshot, let me try again:
Screenshot 2020-11-11 at 9 27 30

the size of the modular window is a main difference

**Comment by vanwagonet** [comment]
Hmm, I'm not sure what caused that, but it doesn't seem like a problem to me. Are you ok with it being smaller?

**Review by balintbartok — APPROVED** [review]
QA+1
